### ISSUE TITLE
Revise MAAS in LXD instructions

### ISSUE THREAD

**Issue by dshcherb** [CanonicalLtd/maas-docs#700]
There is a requirement in the current documentation for a container to be created with the following options:

```
raw.lxc: |-
    lxc.cgroup.devices.allow = c 10:237 rwm
    lxc.aa_profile = unconfined
    lxc.cgroup.devices.allow = b 7:* rwm
  security.privileged: "true"
```

Not sure about aa_profile but device directives are there for managing loop devices (loop-control and /dev/loop<i>) and mounting file systems. Only a root user in the [root user namespace](http://man7.org/linux/man-pages/man7/user_namespaces.7.html) can use [CAP_SYS_ADMIN](http://man7.org/linux/man-pages/man2/mount.2.html) capability which is required to mount a file system.

http://elixir.free-electrons.com/linux/v4.14.2/source/Documentation/admin-guide/devices.txt#L369
```
   7 block	Loopback devices
		  0 = /dev/loop0	First loop device
		  1 = /dev/loop1	Second loop device
		    ...

		The loop devices are used to mount filesystems not
		associated with block devices.	The binding to the
		loop devices is handled by mount(8) or losetup(8).
...
  10 char       Non-serial mice, misc features
...
                237 = /dev/loop-control Loopback control device
```

http://elixir.free-electrons.com/linux/v4.14.2/source/include/uapi/linux/capability.h#L278
```
...
/* Allow mount() and umount(), setting up new smb connection */
...
#define CAP_SYS_ADMIN        21
...
```

I have not found any references to usage of loop devices in 2.3.0 except for "uec2roottar" ones: http://paste.ubuntu.com/26056577/

This was used at some point to extract contents of images:

https://github.com/maas/maas/commit/7e2bff357855a5b1da14fee99389bc535948f340#diff-206173771dc49f98ac3f9aeea660fb25R74

===

If you use snaps on a host, then you use loop devices which immediately triggers a problem with sharing those with a container.

Moreover, using privileged containers has other prlimit-related side-effects: https://github.com/CanonicalLtd/maas-docs/issues/209

Could we drop that requirement from the documentation for newer MAAS versions? As far as I can see, MAAS doesn't use loop devices for relevant image manipulations anymore.

**Comment by degville** [comment]
Thanks for flagging this (and for such a detailed report!). We'll run through the LXD instructions and update accordingly.

**Comment by degville** [comment]
After checking with the MAAS team, the configuration necessity for LXD loop devices is still a requirement and isn't planned to be dropped before release of MAAS 2.4.

**Comment by degville** [comment]
I'm going to close this for now, but we'll keep these details in mind when those changes are landed for 2.4.
